packages/reactive-element/src/decorators/base.ts

```diff
@@ -44,3 +46,66 @@ export const standardPrototypeMethod = (
   key: element.key,
   descriptor,
 });
+
+/**
+ * Helper for decorating a property that is compatible with both TypeScript
+ * and Babel decorators. The optional `finisher` can be used to perform work on
+ * the class. The optional `descriptor` should return a PropertyDescriptor
+ * to install for the given property.
+ *
+ * @param finisher {(ctor: typeof ReactiveElement, property: PropertyKey) => void)} Optional finisher
+ * @param descriptor {(property: PropertyKey) => PropertyDescriptor} Optional descriptor generator
+ * @returns {ClassElement|void}
+ */
+export const decorateProperty = ({
+  finisher,
+  descriptor,
+}: {
+  finisher?:
+    | ((ctor: typeof ReactiveElement, property: PropertyKey) => void)
+    | null;
+  descriptor?: (property: PropertyKey) => PropertyDescriptor;
+}) => (
+  protoOrDescriptor: ReactiveElement | ClassElement,
+  name?: string
+  // Note TypeScript requires the return type to be `void|any`
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): void | any => {
+  // TypeScript / Babel legacy mode
+  if (name !== undefined) {
+    const ctor = (protoOrDescriptor as ReactiveElement)
+      .constructor as typeof ReactiveElement;
+    if (descriptor !== undefined) {
+      Object.defineProperty(protoOrDescriptor, name, descriptor(name));
+    }
+    finisher?.(ctor, name!);
+    // Babel standard mode
+  } else {
+    // Note, the @property decorator saves `key` as `originalKey`
+    // so try to use it here.
+    const key =
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (protoOrDescriptor as any).originalKey ??
+      (protoOrDescriptor as ClassElement).key;
+    const info: ClassElement =
+      descriptor != undefined
+        ? {
+            kind: 'method',
+            placement: 'prototype',
+            key,
+            descriptor: descriptor((protoOrDescriptor as ClassElement).key),
+          }
+        : {...(protoOrDescriptor as ClassElement), key};
+    if (finisher != undefined) {
+      info.finisher = function <ReactiveElement>(
+        ctor: Constructor<ReactiveElement>
+      ) {
+        finisher(
+          (ctor as unknown) as typeof ReactiveElement,
+          (protoOrDescriptor as ClassElement).key
```

Comment:
This doesn't need to be `key` so it gets the `originalKey ?? key` treatment?

Reply:
Fixed.